src/llamafactory/data/parser.py

```diff
@@ -97,20 +97,29 @@ def get_dataset_list(dataset_names: Optional[Sequence[str]], dataset_dir: str) -
 
     dataset_list: List["DatasetAttr"] = []
     for name in dataset_names:
-        if dataset_info is None:  # dataset_dir is ONLINE
-            load_from = "ms_hub" if use_modelscope() else "hf_hub"
+        if dataset_info is None: # dataset_dir is ONLINE
+            load_from = None
+            if use_openmind():
+                load_from = "om_hub"
+            if use_modelscope():
+                load_from = "ms_hub"
+            if load_from is None:
+                load_from = "hf_hub"
             dataset_attr = DatasetAttr(load_from, dataset_name=name)
             dataset_list.append(dataset_attr)
             continue
 
         if name not in dataset_info:
             raise ValueError("Undefined dataset {} in {}.".format(name, DATA_CONFIG))
 
+        has_om_url = "om_hub_url" in dataset_info[name]
         has_hf_url = "hf_hub_url" in dataset_info[name]
         has_ms_url = "ms_hub_url" in dataset_info[name]
 
-        if has_hf_url or has_ms_url:
-            if (use_modelscope() and has_ms_url) or (not has_hf_url):
+        if has_om_url or has_hf_url or has_ms_url:
+            if (use_openmind() and has_om_url) or (not has_hf_url):
```

Comment:
`if has_om_url and (use_openmind() or not has_hf_url)`

Reply:
Bug fixed, thanks for your review!